test/parallel_api/algorithm/alg.sorting/sort_by_key_common.h

```diff
@@ -0,0 +1,177 @@
+
+    // 2. Sorting
+    if constexpr (std::is_same_v<StabilityTag, StableSort>)
+        oneapi::dpl::stable_sort_by_key(policy, keys_begin, keys_begin + n, vals_begin);
+    else
+        oneapi::dpl::sort_by_key(policy, keys_begin, keys_begin + n, vals_begin);
+}
+
+template<typename _Keys, typename _Vals, typename _Size>
+void
+check_sort_by_key_results(const _Keys& keys_buf, const _Vals& vals_buf,  _Size n)
+{
+    //Checking results
+    const int k = (n - 1) / 2 + 1;
+    for (int i = 0; i < n; ++i)
+    {
+        if(i - k < 0)
+        {
+             //a key should be 1 and value should be even
+            EXPECT_TRUE(keys_buf[i] == 1 && vals_buf[i] % 2 == 0, "wrong sort_by_key result with a standard policy");
+        }
+        else
+        {
+            //a key should be 2 and value should be odd
+            EXPECT_TRUE(keys_buf[i] == 2 && vals_buf[i] % 2 == 1, "wrong sort_by_key result with a standard policy");
+        }
+    }
+}
+
+template<typename _Policy, typename StabilityTag>
+void
+test_with_std_policy(_Policy&& policy, StabilityTag stability_tag)
+{
+    constexpr int n = 1000000;
+    std::vector<int> keys_buf(n); //keys
+    std::vector<int> vals_buf(n); //values
+
+    call_sort_by_key(policy, keys_buf.begin(), vals_buf.begin(), n);
+    check_sort_by_key_results(keys_buf, vals_buf, n, stability_tag);
+}
+
+#if TEST_DPCPP_BACKEND_PRESENT
+
+#include "support/sycl_alloc_utils.h"
+
+template <sycl::usm::alloc alloc_type, typename StabilityTag>
+void
+test_with_usm(sycl::queue& q, StabilityTag)
+{
+    constexpr int N = 32;
+
+    int h_key[N] = {};
+    int h_val[N] = {};
+    for (int i = 0; i < N; i++)
+    {
+        h_val[i] = ((N - 1 - i) / 3) * 3;
+        h_key[i] = i * 10;
+    }
+
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper_h_key(q, ::std::begin(h_key), ::std::end(h_key));
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper_h_val(q, ::std::begin(h_val), ::std::end(h_val));
+
+    int* d_key = dt_helper_h_key.get_data();
+    int* d_val = dt_helper_h_val.get_data();
+
+    auto myPolicy = TestUtils::make_device_policy<
+        TestUtils::unique_kernel_name<class copy, TestUtils::uniq_kernel_index<alloc_type>()>>(q);
+
+    if constexpr (std::is_same_v<StabilityTag, StableSort>)
+        oneapi::dpl::stable_sort_by_key(myPolicy, d_key, d_key + N, d_val, std::greater<void>());
+    else
+        oneapi::dpl::sort_by_key(myPolicy, d_key, d_key + N, d_val, std::greater<void>());
+
+    int h_skey[N] = {};
+    int h_sval[N] = {};
+
+    dt_helper_h_key.retrieve_data(h_skey);
+    dt_helper_h_val.retrieve_data(h_sval);
+
+    for (int i = 0; i < N; i++)
+    {
+        if (i < (N - 1))
+        {
+            EXPECT_TRUE(h_skey[i] >= h_skey[i + 1], "wrong sort result with hetero policy, USM data");
+        }
+    }
+}
+
+template <typename StabilityTag>
+void
+test_with_buffers(sycl::queue& q, StabilityTag stability_tag)
```

Comment:
Early this function was written under `#if TEST_DPCPP_BACKEND_PRESENT`
Why this define is not used now?

Reply:
It is used to guard  `test_with_buffers` and `test_usm` as before. I additionally guarded `test_all_policies` by mistake, but is fixed now.